File: passkey-app/src/main/java/com/crosscert/passkey/app/api/v1/rp/dto/RegistrationFinishRequest.java
package com.crosscert.passkey.app.api.v1.rp.dto;

import com.fasterxml.jackson.databind.JsonNode;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record RegistrationFinishRequest(
        @NotBlank String registrationToken,
        @NotNull JsonNode publicKeyCredential
) {}

File: sdk-java/src/main/kotlin/com/crosscert/passkey/sdk/dto/RegistrationFinishRequest.kt
package com.crosscert.passkey.sdk.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.JsonNode

data class RegistrationFinishRequest(
    @get:JvmName("registrationToken") @get:JsonProperty("registrationToken") @param:JsonProperty("registrationToken") val registrationToken: String,
    @get:JvmName("publicKeyCredential") @get:JsonProperty("publicKeyCredential") @param:JsonProperty("publicKeyCredential") val publicKeyCredential: JsonNode,
)
